Update: FModel Docs (#270)
satisfactorymodding/Documentation: modules/ROOT/pages/Development/ExtractGameFiles.adoc

=== Specify Custom Version and Mapping

Unreal Engine 5 introduced changes in how assets are stored in pak files.
You will need to provide FModel with extra information so that it can read the data correctly.

> ==== The required files can be found in your game install directory at

Reviewer: Don't use headings for emphasis. Also, made this a link to the FAQ item
```suggestion
The required files can be found in your
xref:faq.adoc#Files_GameInstall[game install directory] as
```

`/CommunityResources/FactoryGame.usmap`
and `/CommunityResources/CustomVersions.json`. 

To start navigate to `Settings` > `General`.
In the "UE Versions" dropdown, select `GAME_UE5_2 (536870944)`,
then press the `Custom Versions` button in the "Version Configuration" row.

> [WARNING]

Reviewer: This warning is placed in a somewhat confusing order because it shows up before the file you're being asked to edit. I'll reword this section after it gets merged, just leaving a comment here as a note

Author: The idea was that people would see the emphasized bit about UE5 changes and file locations.
Based on this conversation: https://discord.com/channels/555424930502541343/601030071221878784/1190644182923088015

In regards to the warning at Line 57, they were introduced to the file names and locations at the beginning of this at Line 46. And the warning about Win10 vs Win11 should be just before they are editing the custom versions, no?

====
If you are using Windows 10, Notepad loads the .json as UTF-16 LE which will work with FModel.

If you are using Windows 11, you will need to use an application such as https://notepad-plus-plus.org/[Notepad++]. 
Windows 11 Notepad loads the .json in UTF-8 and FModel will tell you that you broke the .json if you copy the contents over.

image:ExtractingGameFiles/FModel_BrokenJSON.png[Broken JSON]
====

Delete the text contents of the popup window that appears
and replace it with the contents of the `CustomVersions.json` file from the location listed above 

> (for example, `C:\EpicGamesGames\SatisfactoryExperimental\CommunityResources\CustomVersions.json`) ,

Reviewer: Remove extra space
```suggestion
(for example, `C:\EpicGamesGames\SatisfactoryExperimental\CommunityResources\CustomVersions.json`),
```

then press `OK` to save changes.

Back on the Settings screen, check the box for "Local Mapping File"

> and then enter the full path to your `FactoryGame.usmap` file 

Reviewer: Remove random extra space
```suggestion
and then enter the full path to your `FactoryGame.usmap` file
```
